extensions/azurecore/src/azureResource/services/terminalService.ts

```diff
@@ -168,13 +168,14 @@ class AzureTerminal implements vscode.Pseudoterminal {
 		this.socket.removeAllListeners('close');
 
 		this.socket.terminate();
+		this.socket.close();
 		if (this.intervalTimer) {
 			clearInterval(this.intervalTimer);
 		}
 	}
 
 	async setDimensions(dimensions: vscode.TerminalDimensions): Promise<void> {
-		if (!dimensions) {
+		if (!dimensions || this.terminalDimensions) {
```

Comment:
And what happens if the terminal dimensions actually change? We would just be ignoring that.

Is setDimensions being called multiple times with the same row/column values? If so then I think the better fix would be to only call resetTerminalSize if one of those values are different.